### PR TITLE
use configError over validationErrorMessage

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- only show FE validation error message if there's no message provided by the API 
- this includes on change to the helm test. requires merging testim branch after this is merged 

#### Which issue(s) this PR fixes:

Fixes # https://app.shortcut.com/replicated/story/75670/errors-when-saving-config-are-not-propagated-to-ui

#### Special notes for your reviewer:

- shows the backend error when there is one. otherwise it falls back to the FE supplied error related to validation 

Screenshot 2023-05-02 at 11 29 06
Screenshot 2023-05-02 at 11 29 17

#### Does this PR introduce a user-facing change?

```release-note
Config page will now show the correct error message for errors not related to validation. 
```

#### Does this PR require documentation?

None

### CODE REVIEW

File: web/src/features/AppConfig/components/AppConfig.tsx [hunk 1]
```diff
@@ -76,18 +76,19 @@ type State = {
...
   configGroups: ConfigGroup[];
   configLoading: boolean;
   displayErrorModal: boolean;
   downstreamVersion: Version | null;
   errorTitle: string;
   gettingConfigErrMsg: string;
-  showValidationError: boolean;
   initialConfigGroups: ConfigGroup[];
   savingConfig: boolean;
+  showConfigError: boolean;
```
Comment (rianfowler) [inline web/src/features/AppConfig/components/AppConfig.tsx:88]: `configError` -> `showConfigError` - probably don't need both of these but it's more explicit to have both.

Comment (miaawong) [reply web/src/features/AppConfig/components/AppConfig.tsx:88]: it's good to be more explicit, easier to read imo

File: web/src/features/AppConfig/components/AppConfig.tsx [hunk 2]
```diff
@@ -565,7 +582,7 @@ class AppConfig extends Component<Props, State> {
           );
 
         this.setState({
-          configError: hasValidationError,
+          // configError: hasValidationError,
```
Comment (miaawong) [inline web/src/features/AppConfig/components/AppConfig.tsx:585]: is this comment intended?

File: web/src/features/AppConfig/components/AppConfig.tsx [hunk 3]
```diff
@@ -725,6 +746,9 @@ class AppConfig extends Component<Props, State> {
       observer.observe(section);
     });
 
+    console.log("configErrorMessage", configErrorMessage);
```
Comment (miaawong) [inline web/src/features/AppConfig/components/AppConfig.tsx:749]: are these console.logs needed?

**Review by miaawong — APPROVED** [review]

File: web/src/features/AppConfig/components/AppConfig.tsx (hunk 3 above)
Comment (rianfowler) [reply web/src/features/AppConfig/components/AppConfig.tsx:749]: whoops

**Pull request merged**